clang/lib/Lex/LiteralSupport.cpp

```diff
@@ -1241,7 +1260,9 @@ bool NumericLiteralParser::isValidUDSuffix(const LangOptions &LangOpts,
     return false;
 
   // By C++11 [lex.ext]p10, ud-suffixes starting with an '_' are always valid.
-  if (Suffix[0] == '_')
+  // Suffixes starting with '__' (double underscore) are for use by
+  // implementation.
+  if (Suffix[0] == '_' && Suffix[1] != '_')
```

Comment:
Is there a code path where we get here with the `Suffix` only being `1` character long (e.g. if we have just `1234_` maybe)? Because this will assert on an out-of-bounds access if that ever happens; if that’s impossible, then this is fine, but if you can find a case that triggers this, then you might consider using `Suffix.starts_with` here (and definitely also add it as a test case in that case).

Reply:
Example code that causes an assertion now:
```
unsigned operator ""_(const char *);

unsigned i = 1_;
```